File: aws-node-rest-api-with-dynamodb/todos/get.js
'use strict';

const AWS = require('aws-sdk'); // eslint-disable-line import/no-extraneous-dependencies

const dynamoDb = new AWS.DynamoDB.DocumentClient();

module.exports.get = (event, context, callback) => {
  const params = {
    TableName: process.env.DYNAMODB_TABLE,
    Key: {
      id: event.pathParameters.id,
    },
  };

  // fetch todo from the database
  dynamoDb.get(params, (error, result) => {
    // handle potential errors
    if (error) {
      console.error(error);
      callback(new Error('Couldn\'t fetch the todo item.'));
      return;
    }

    // create a response
    const response = {
      statusCode: 200,
      body: JSON.stringify(result.Item),
    };
    callback(null, response);
  });
};

File: aws-python-rest-api-with-faunadb/todos/get.py
import json

from todos.makeresult import make_result
from todos import client, TODOS

from faunadb.objects import Ref
from faunadb import query

def get(event, context):
    # fetch todo from the database
    ref = Ref(TODOS, event['pathParameters']['id'])
    fetched = client.query(query.get(ref))

    # create a response
    response = {
        "statusCode": 200,
        "body": json.dumps(make_result(fetched))
    }

    return response
